### PR TITLE
Support T-SQL sp_enum_oledb_providers stored procedure

### PR DESCRIPTION
**Pull request by thephantomthief**
### Description

This commit adds support for the T-SQL sp_enum_oledb_providers stored procedure. This procedure shows the list of available providers in the babelfish server. Since, we only support TDS servers as linked servers via the tds_fdw extension, this procedure will show tds_fdw as the available provider if installed.

Task: BABEL-4273
Co-authored-by: Shriramu A R <shriraar@amazon.com>
Co-authored-by: Sharu Goel <goelshar@amazon.com>
Signed-off-by: Sharu Goel <goelshar@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/plerrcodes.h [hunk 1]
```diff
@@ -1006,4 +1006,4 @@
 
 {
 	"pltsql_error_not_mapped", ERRCODE_PLTSQL_ERROR_NOT_MAPPED
-},
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/plerrcodes.h]: unwanted change

File: contrib/babelfishpg_tsql/src/procedures.c [hunk 2]
```diff
@@ -3749,4 +3751,73 @@ remove_delimited_identifer(char *str)
...
+	SetRemoteDestReceiverParams(receiver, portal);
+
+	/* fetch the result and return the result-set */
+	PortalRun(portal, FETCH_ALL, true, true, receiver, receiver, NULL);
+
+	receiver->rDestroy(receiver);
+
+	SPI_cursor_close(portal);
+
+	if ((rc = SPI_finish()) != SPI_OK_FINISH)
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/procedures.c]: do we not need any error handling here? will we handle errors gracefully in any of the above steps?

Comment (thephantomthief) [reply contrib/babelfishpg_tsql/src/procedures.c]: Surrounded with TRY...CATCH block

File: contrib/babelfishpg_tsql/src/plerrcodes.h (hunk 1 above)
Comment (thephantomthief) [reply contrib/babelfishpg_tsql/src/plerrcodes.h]: done

**Review by surendravishnoi — APPROVED** [review]

**Review by basasairohan — APPROVED** [review]

**Pull request merged**